### PR TITLE
Start removing implicit any errors

### PR DESCRIPTION
**Pull request by smfoote**
Thusfar:

component-managers/abstract.ts (2 errors remain)
component-managers/curly.ts (some errors remain)
component-managers/mount.ts
utils/curly-component-state-bucket.ts

### CODE REVIEW

File: packages/container/lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+interface Container {
```
Comment (krisselden) [inline packages/container/lib/index.d.ts]: I stubbed out a more limited interface for this in ember-glimmer we should unify

File: packages/ember-glimmer/lib/component-managers/curly.ts [hunk 2]
```diff
@@ -74,26 +91,26 @@ function applyAttributeBindings(element, attributeBindings, component, operation
...
+function ariaRole(vm: any) {
   return vm.getSelf().get('ariaRole');
 }
 
 class CurlyComponentLayoutCompiler {
   static id: string;
   public template: any;
 
-  constructor(template) {
+  constructor(template: any) {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/component-managers/curly.ts]: I did make a type for template

File: packages/ember-glimmer/lib/component-managers/curly.ts [hunk 3]
```diff
@@ -74,26 +91,26 @@ function applyAttributeBindings(element, attributeBindings, component, operation
...
-function tagName(vm) {
+function tagName(vm: any) {
   // tslint:disable-next-line:no-shadowed-variable
   let { tagName } = vm.dynamicScope().view;
 
   return PrimitiveReference.create(tagName === '' ? null : tagName || 'div');
 }
 
-function ariaRole(vm) {
+function ariaRole(vm: any) {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/component-managers/curly.ts]: there is a type exported for this, I think it is called PublicVM

File: packages/ember-glimmer/lib/component-managers/curly.ts [hunk 4]
```diff
@@ -107,22 +124,22 @@ export class PositionalArgumentReference {
...
     this._references = references;
   }
 
   value() {
-    return this._references.map((reference) => reference.value());
+    return this._references.map((reference: any) => reference.value());
   }
 
-  get(key) {
+  get(key: any) {
```
Comment (krisselden) [inline packages/ember-glimmer/lib/component-managers/curly.ts]: string

File: packages/ember-glimmer/lib/component-managers/curly.ts [hunk 5]
```diff
@@ -231,16 +248,15 @@ export default class CurlyComponentManager extends AbstractManager<ComponentStat
...
-      let { component } = bucket;
-      template = this.templateFor(component, env);
+      template = this.templateFor(bucket, env);
     }
     return env.getCompiledBlock(CurlyComponentLayoutCompiler, template);
   }
 
-  templateFor(component, env) {
+  templateFor(component: ComponentStateBucket, env: Environment): any {
     let Template = get(component, 'layout');
```
Comment (krisselden) [inline packages/ember-glimmer/lib/component-managers/curly.ts]: we have a type for what templates return look at the template .d.ts I updated

File: packages/ember-routing/lib/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+export const Location: any;
+export const NoneLocation: any;
+export const HashLocation: any;
+export const HistoryLocation: any;
+export const AutoLocation: any;
+
```
Comment (krisselden) [inline packages/ember-routing/lib/index.d.ts]: lets at least type the signature of the functions

File: packages/ember-glimmer/lib/component-managers/curly.ts (hunk 3 above)
Comment (smfoote) [reply packages/ember-glimmer/lib/component-managers/curly.ts]: Works for the `ariaRole` function, but not `tagName`: `Property 'view' does not exist on type 'DynamicScope'`

**Comment by smfoote** [comment]
Despite the green check, I don't think this is ready yet. There are several TypeScript errors still.

**Comment by rwjblue** [comment]
@smfoote - Sounds good, let us know when you think it is ready for another round of review and possible merging.

**Comment by smfoote** [comment]
Since the checks are passing, we might merge this one as is then fix the remaining TypeScript errors in a future PR. I'll leave that up to @krisselden

**Pull request merged**